File: Java/26_RemoveDuplicates/Solution.java

public class Solution {
  public int removeDuplicates (int[] numbers) {
    if ( numbers.length == 0) return 0;
    int aux = 1;
    for (int i = 1; i < numbers.length; i++) 
      if (numbers[i-1] != numbers[i]) {
          numbers[aux] = numbers[i];
          aux++;
      }
    return aux;
  }
  public static void main(String [] args) {
    int[] numbers = {0,0,1,1,1,2,2,3,3,4};
    Solution solution = new Solution();
    int n = solution.removeDuplicates(numbers);
    System.out.println(n);
  }
}

File: C++/26_RemoveDuplicates/Solution.cpp
#include<iostream>
#include<vector>

int removeDuplicates (std::vector<int>& nums) {
  if(nums.size() == 0) return 0;
  int aux = 1;
  for ( int i = 1; i < nums.size(); i++)
    if (nums[i-1] != nums[i]) {
      nums[aux] = nums[i];
      aux++;
    }
    return aux;
}

int main (void) {
  std::vector<int> nums = {1,1,2};
  std::cout<<removeDuplicates(nums)<<std::endl;;
  return 0;
}